src/shared/Core/EnvironmentBase.cs

```diff
@@ -104,6 +103,38 @@ public virtual Process CreateProcess(string path, string args, bool useShellExec
 
             return new Process { StartInfo = psi };
         }
+
+        public bool TryLocateExecutable(string program, out string path)
```

Comment:
Awesome! This method is super easy to understand, and it's great we were able to consolidate the UNIX and Windows logic.